visualization/app/codeCharta/util/fileValidator.ts

```diff
@@ -32,42 +32,50 @@ export const ERROR_MESSAGES = {
 	blacklistError: "Excluding all buildings is not possible."
 }
 
-export function validate(nameDataPair: NameDataPair) {
-	const file = nameDataPair?.content
-	const result: CCValidationResult = { error: [], warning: [] }
+export function checkWarnings(file: ExportCCFile) {
+	const warnings: string[] = []
+	if (!file) {
+		return warnings
+	}
+	if (fileHasHigherMinorVersion(file)) {
+		warnings.push(`${ERROR_MESSAGES.minorApiVersionOutdated} Found: ${file.apiVersion}`)
+	}
+	return warnings
+}
+
+export function checkErrors(file: ExportCCFile) {
+	const errors: string[] = []
 	switch (true) {
 		case !file:
-			result.error.push(ERROR_MESSAGES.fileIsInvalid)
+			errors.push(ERROR_MESSAGES.fileIsInvalid)
 			break
 		case !isValidApiVersion(file):
-			result.error.push(ERROR_MESSAGES.apiVersionIsInvalid)
+			errors.push(ERROR_MESSAGES.apiVersionIsInvalid)
 			break
 		case fileHasHigherMajorVersion(file):
-			result.error.push(ERROR_MESSAGES.majorApiVersionIsOutdated)
-			break
-		case fileHasHigherMinorVersion(file):
-			result.warning.push(`${ERROR_MESSAGES.minorApiVersionOutdated} Found: ${file.apiVersion}`)
+			errors.push(ERROR_MESSAGES.majorApiVersionIsOutdated)
 			break
 	}
+	if (errors.length === 0) errors.push(...checkJsonSchema(file))
+	return errors
+}
 
-	if (result.error.length === 0) {
+function checkJsonSchema(file: ExportCCFile) {
+	const errors: string[] = []
+	if (errors.length === 0) {
```

Comment:
This is always true, isn't it?

Reply:
Missed that! I will clean it up, thanks 😃

Reply:
Fixed in next commit